docs/sources/alerting/learn/examples/multi-dimensional-alerts.md

```diff
@@ -0,0 +1,149 @@
+---
+
+# Example of multi-dimensional alerts on time series data
+
+This example shows how a single alert rule can generate multiple alert instances — one for each label set (or time series). This is called **multi-dimensional alerting**: one alert rule, many alert instances.
+
+In Prometheus, each unique combination of labels defines a distinct time series. Grafana Alerting uses the same model: each label set is evaluated independently, and a separate alert instance is created for each series.
+
+This pattern is common in dynamic environments when monitoring a group of components like multiple CPUs, containers, or per-host availability. Instead of defining individual alert rules or aggregated alerts, you alert on _each dimension_ — so you can detect particular issues and include that level of detail in notifications.
+
+For example, a query returns one series per CPU:
+
+| `cpu` label value | CPU percent usage |
+| :---------------- | :---------------- |
+| cpu-0             | 95                |
+| cpu-1             | 30                |
+| cpu-2             | 85                |
+
+With a threshold of `> 80`, this would trigger two alert instances for `cpu-0` and one for `cpu-2`.
+
+## Examples overview
+
+Imagine you want to trigger alerts when CPU usage goes above 80%, and you want to track each CPU core independently.
+
+You can use a Prometheus query like this:
+
+```
+sum by(cpu) (
+  rate(node_cpu_seconds_total{mode!="idle"}[1m])
+)
+```
+
+This query returns the active CPU usage rate per CPU core, averaged over the past minute.
+
+| CPU core | Active usage rate |
+| :------- | :---------------- |
+| cpu-0    | 95                |
+| cpu-1    | 30                |
+| cpu-2    | 85                |
+
+This produces one series for each existing CPU.
+
+When Grafana Alerting evaluates the query, it creates an individual alert instance for each returned series.
+
+| Alert instance | Value |
+| :------------- | :---- |
+| {cpu="cpu-0"}  | 95    |
+| {cpu="cpu-1"}  | 30    |
+| {cpu="cpu-2"}  | 85    |
+
+With a threshold condition like `$A > 80`, Grafana evaluates each instance separately and fires alerts only where the condition is met:
+
+| Alert instance | Value | State  |
+| :------------- | :---- | :----- |
+| {cpu="cpu-0"}  | 95    | Firing |
+| {cpu="cpu-1"}  | 30    | Normal |
+| {cpu="cpu-2"}  | 85    | Firing |
+
+Multi-dimensional alerts help you surface issues on individual components—problems that might be missed when alerting on aggregated data (like total CPU usage). Each alert instance includes distinct label data that identifies the component, making the alert more specific and actionable.
+
+For example, Grafana might generate alerts like:
+
+- High CPU usage on `cpu-0`
+- High CPU usage on `cpu-2`
+
+{{< admonition type="note" >}}
+To avoid sending multiple notifications for similar alerts, Grafana Alerting [groups similar immediate alerts](ref:alert-grouping) into a single notification by default.
```

Comment:
It's not clear what you mean when you say "immediate"

Reply:
addressed by https://github.com/grafana/grafana/pull/105325/commits/1751830f0feb013140707a37f43fb45005eca190

I decided removing the note about alert grouping to simplify this example which displays an annotation template with the recent changes.